stack/stack.go

```diff
@@ -32,6 +32,13 @@ func ParseYAML(yamlFile string) (*Services, error) {
 			return nil, err
 		}
 	}
+	return ParseYamlData(fileData, filter)
+}
+
+// ParseYamlData parse YAML data into a stack of "services".
+func ParseYamlData(fileData []byte, filter string) (*Services, error) {
+	var services Services
+	var err error
 
 	err = yaml.Unmarshal(fileData, &services)
```

Comment:
can just use a short assignment here `:=` and dump the `err` forward declaration on line 41.